scio-core/src/main/scala/com/spotify/scio/coders/Coder.scala

```diff
@@ -132,9 +132,39 @@ private final case class DisjunctionCoder[T, Id](typeName: String,
 // XXX: Workaround a NPE deep down the stack in Beam
 // info]   java.lang.NullPointerException: null value in entry: T=null
 private case class WrappedBCoder[T](u: BCoder[T]) extends BCoder[T] {
+
+  /**
+   * Eagerly compute a stack trace on materialization
+   * to provide a helpful stacktrace if an exception happens
+   */
+  private val stackTrace: Array[StackTraceElement] =
+    Thread
+      .currentThread()
+      .getStackTrace()
+      .dropWhile(_.getClassName.contains(WrappedBCoder.getClass.getName))
+
+  private def buildException(cause: Throwable): Exception =
+    new RuntimeException {
```

Comment:
@jto should we provide this as `CoderException` instead of just `RuntimeException`?